features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/package.json

```diff
@@ -27,6 +27,7 @@
     "material-ui-icons": "^1.0.0-alpha.19",
     "prop-types": "^15.5.10",
     "qs": "^6.4.0",
+    "rc-notification": "^3.0.1",
```

Comment:
Why we need this?

Reply:
@dakshika 
This has been used in writing a messaging library for MUI [Snackbar](https://material-ui-next.com/demos/snackbars/),
Previously with `antd` library we have used [message](https://ant.design/components/message/) feature of it to display messages to UI,Since we are moving to MUI, we had to find a messaging library (because MUI Snackbar implementation doesn't give OOB messaging/alerting capabilities), So as a quick workaround we adopt [react-notification-system](https://github.com/igorprado/react-notification-system) by following the ballerina [composer usage](https://github.com/ballerinalang/composer/blob/master/modules/web/package.json#L77) of it https://github.com/wso2/product-apim/issues/1950,
But it doesn't give sleek UX to the user since `react-notification-system` doesn't follow material design concept and when we are using it breack the API Manager material based ux.
So that we build a notification library for MUI using what `antd` [message](https://github.com/ant-design/ant-design/blob/master/components/message/index.tsx) library has used for their messaging feature
[rc-notification](https://github.com/react-component/notification)

With this implementation, we could show following types of alerts to the user

![alerts](https://user-images.githubusercontent.com/3313885/35136517-16b168de-fd0a-11e7-83d0-cd1b33edfc41.gif)

source code for above alert messages is as below

```javascript
import Alert from '../Shared/Alert'
.
. .
        Alert.error("Somthing went wrong!");
        Alert.info("You are done!");
        Alert.warning("Is everything correct?");
        Alert.loading("Still loading data . . .");
        
```

Just like the `antd` messaging feature, no need to worry about putting snack bar component inside `render()` method.

![image](https://user-images.githubusercontent.com/3313885/35136978-ccb0c9b6-fd0c-11e7-97ff-955129f106ef.png)

how [rc-notification](https://github.com/react-component/notification) does that is, they create a new react dom root , inside a div container which is appended to app's html container and keep popping messages through that